File: HW/HW8/HW8.py
from collections import Counter

class Solution:
    def longestPalindrome(self, s: str) -> int:
        frequency_map = Counter(s)
        length = 0
        has_odd_frequency = False
        
        for count in frequency_map.values():
            length += count // 2 * 2
            if count % 2 == 1:
                has_odd_frequency = True
        
        if has_odd_frequency:
            length += 1
        
        return length

File: HW/HW8/HW8.java
import java.util.HashMap;
import java.util.Map;

public class HW8 {
    public int longestPalindrome(String s) {
        Map<Character, Integer> frequencyMap = new HashMap<>();
        for (char c : s.toCharArray()) {
            frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
        }
        int length = 0;
        boolean hasOddFrequency = false;
        for (int count : frequencyMap.values()) {
            length += count / 2 * 2;
            if (count % 2 == 1) {
                hasOddFrequency = true;
            }
        }
            if (hasOddFrequency) {
            length++;
        }
        
        return length;
    }
}
